File: 100 days of leetcode/Day11.cpp
class Solution {
public:
    string reverseVowels(string s) {
        vector<char> vowstr;
        vector<char> vowel {'a','e','i','o','u','A','E','I','O','U'};
        int i = 0, j = s.length() - 1;
        while(i < j){
            while(i < j && find(vowel.begin(), vowel.end(), s[i]) == vowel.end()){
                i++; // Moving i forward until a voewl is found
            }
            while(i < j && find(vowel.begin(), vowel.end(), s[j]) == vowel.end()){
                j--; // Moving j backward until a vowel is found
            }

            if(i < j){
                char temp = s[i];
                s[i] = s[j];
                s[j] = temp;

                i++;
                j--;
            }
            
        }

        return s;
    }
};

File: 100 days of leetcode/Day11.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int i = 0, j = s.length() - 1;
        String vowels = "aeiouAEIOU";

        while (i < j) {
            while (i < j && vowels.indexOf(chars[i]) == -1) {
                i++; // Move i forward until a vowel is found.
            }
            while (i < j && vowels.indexOf(chars[j]) == -1) {
                j--; // Move j backward until a vowel is found.
            }

            if (i < j) {
                char temp = chars[i];
                chars[i] = chars[j];
                chars[j] = temp;
                i++;
                j--;
            }
        }

        return new String(chars);
    }
}
